pkg/jenkinsfile/gitresolver/buildpack.go

```diff
@@ -35,3 +41,29 @@ func InitBuildPack(gitter gits.Gitter, packURL string, packRef string) (string,
 	}
 	return filepath.Join(dir, "packs"), err
 }
+
+func ensureBranchTracksOrigin(dir string, packRef string, gitter gits.Gitter) error {
```

Comment:
just wondering if this should be part of the CloneOrPull function?

Reply:
I was thinking about that, too. I decided to do it ad hoc for these two issues because I don't know what impact it could have in the whole system right now and those issues were critical, but I think we should consider moving it.

Reply:
I've just seen a similar problem when doing `jx upgrade platform` hence why I think this would be good in the CloneOrPull function. I can't see it doing causing any major problems and we could fix forward if we see them.

However we could do it in a two stage approach, merge this and then raise another change to refactor it. That way we'll close off the two critical issues whilst giving us a bit more time to test the wider impact?

Reply:
I think this would be a good approach, I just want to play it safe with this, then we can create an issue to make the general solution.